diff --git a/CMakeLists.txt b/CMakeLists.txt
index 743683f..65efe39 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,6 +3,11 @@
 cmake_minimum_required(VERSION 2.8)
 project (airspy_all)
 
+#provide missing strtoull() for VC11
+if(MSVC11)
+    add_definitions(-Dstrtoull=_strtoui64)
+endif(MSVC11)
+
 add_subdirectory(libairspy)
 add_subdirectory(airspy-tools)
 
@@ -25,14 +30,18 @@ add_custom_target(uninstall
 ########################################################################
 
 if(WIN32 AND NOT CMAKE_CROSSCOMPILING)
+ if (NOT DEFINED LIBUSB_LIBRARIES)
  configure_file(
      ${CMAKE_CURRENT_BINARY_DIR}/../libs_win32/libusb-1.0.dll
      ${CMAKE_CURRENT_BINARY_DIR}/airspy-tools/src/libusb-1.0.dll
  COPYONLY)
+ endif()
 
+ if (NOT DEFINED THREADS_PTHREADS_WIN32_LIBRARY)
  configure_file(
      ${CMAKE_CURRENT_BINARY_DIR}/../libs_win32/pthreadGC2.dll
      ${CMAKE_CURRENT_BINARY_DIR}/airspy-tools/src/pthreadGC2.dll
  COPYONLY)
+ endif()
 endif()
 
diff --git a/libairspy/src/iqconverter_float.c b/libairspy/src/iqconverter_float.c
index 59c160b..844e846 100644
--- a/libairspy/src/iqconverter_float.c
+++ b/libairspy/src/iqconverter_float.c
@@ -46,7 +46,7 @@ THE SOFTWARE.
   #define FIR_STANDARD
   //#define FIR_AUTO_VECTOR
 #else
-	#if (_MSC_VER >= 1300)
+	#if (_MSC_VER >= 1800)
 		#define USE_SSE2
 		#include <immintrin.h>
 	#else
