diff --git a/host/libraries/libbladeRF/src/board/bladerf1/bladerf1.c b/host/libraries/libbladeRF/src/board/bladerf1/bladerf1.c
index 6a78deaa..586ad5a1 100644
--- a/host/libraries/libbladeRF/src/board/bladerf1/bladerf1.c
+++ b/host/libraries/libbladeRF/src/board/bladerf1/bladerf1.c
@@ -348,28 +348,6 @@ static const struct bladerf_lms_port_name_map bladerf1_rx_port_map[] = {
     },
 };
 
-static const struct bladerf_lms_port_name_map bladerf1_tx_port_map[] = {
-/* TODO: TX PA control is not implemented */
-#if 0
-    {
-        FIELD_INIT(.name, "aux"),
-        FIELD_INIT(.tx_pa, PA_AUX),
-    },
-    {
-        FIELD_INIT(.name, "pa1"),
-        FIELD_INIT(.tx_pa, PA_1),
-    },
-    {
-        FIELD_INIT(.name, "pa2"),
-        FIELD_INIT(.tx_pa, PA_2),
-    },
-    {
-        FIELD_INIT(.name, "none"),
-        FIELD_INIT(.tx_pa, PA_NONE),
-    },
-#endif  // 0
-};
-
 /******************************************************************************/
 /* Low-level Initialization */
 /******************************************************************************/
@@ -2050,8 +2028,8 @@ static int bladerf1_set_rf_port(struct bladerf *dev,
     }
 
     if (BLADERF_CHANNEL_IS_TX(ch)) {
-        port_map     = bladerf1_tx_port_map;
-        port_map_len = ARRAY_SIZE(bladerf1_tx_port_map);
+        port_map     = NULL;
+        port_map_len = 0;
     } else {
         port_map     = bladerf1_rx_port_map;
         port_map_len = ARRAY_SIZE(bladerf1_rx_port_map);
@@ -2121,8 +2099,8 @@ static int bladerf1_get_rf_port(struct bladerf *dev,
     }
 
     if (BLADERF_CHANNEL_IS_TX(ch)) {
-        port_map     = bladerf1_tx_port_map;
-        port_map_len = ARRAY_SIZE(bladerf1_tx_port_map);
+        port_map     = NULL;
+        port_map_len = 0;
 
 #if 0
         status = lms_get_pa(dev, &tx_pa);
@@ -2178,8 +2156,8 @@ static int bladerf1_get_rf_ports(struct bladerf *dev,
     size_t i;
 
     if (BLADERF_CHANNEL_IS_TX(ch)) {
-        port_map     = bladerf1_tx_port_map;
-        port_map_len = ARRAY_SIZE(bladerf1_tx_port_map);
+        port_map     = NULL;
+        port_map_len = 0;
     } else {
         port_map     = bladerf1_rx_port_map;
         port_map_len = ARRAY_SIZE(bladerf1_rx_port_map);
