diff --git a/host/libraries/libbladeRF_test/CMakeLists.txt b/host/libraries/libbladeRF_test/CMakeLists.txt
index 48f93d94..34afa6a7 100644
--- a/host/libraries/libbladeRF_test/CMakeLists.txt
+++ b/host/libraries/libbladeRF_test/CMakeLists.txt
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 2.8)
 add_subdirectory(test_async)
 add_subdirectory(test_bootloader_recovery)
 add_subdirectory(test_c)
-add_subdirectory(test_config_file)
+#add_subdirectory(test_config_file)
 add_subdirectory(test_cpp)
 add_subdirectory(test_ctrl)
 add_subdirectory(test_freq_hop)
