diff --git a/host/common/include/host_config.h.in b/host/common/include/host_config.h.in
index 7c6ca5d..0ef97dd 100644
--- a/host/common/include/host_config.h.in
+++ b/host/common/include/host_config.h.in
@@ -162,8 +162,12 @@
 #if _MSC_VER
 #   define strtok_r    strtok_s
 #   define strtoull    _strtoui64
-#   define snprintf    _snprintf
-#   define vsnprintf   _vsnprintf
+#if _MSC_VER < 1900
+	#   define snprintf    _snprintf
+	#   define vsnprintf   _vsnprintf
+#else
+	#	define STDC99
+#endif
 #   define strcasecmp  _stricmp
 #   define strncasecmp _strnicmp
 #   define fileno      _fileno
