diff --git a/host/common/src/dc_calibration.c b/host/common/src/dc_calibration.c
index 4ee969e6..e9ac4b21 100644
--- a/host/common/src/dc_calibration.c
+++ b/host/common/src/dc_calibration.c
@@ -676,8 +676,8 @@ static int rx_cal_dc_off(struct rx_cal *cal, struct gain_mode *gains,
     }
 
     sample_mean(cal->samples, cal->num_samples, &mean_i, &mean_q);
-    *dc_i = mean_i;
-    *dc_q = mean_q;
+    *dc_i = (int16_t)mean_i;
+    *dc_q = (int16_t)mean_q;
 
     return 0;
 }
