diff --git a/gr-blocks/lib/udp_source_impl.cc b/gr-blocks/lib/udp_source_impl.cc
index b1977a8..6d3bd7c 100644
--- a/gr-blocks/lib/udp_source_impl.cc
+++ b/gr-blocks/lib/udp_source_impl.cc
@@ -94,7 +94,14 @@ namespace gr {
         d_socket->open(d_endpoint.protocol());
 
         boost::asio::socket_base::linger loption(true, 0);
-        d_socket->set_option(loption);
+        try
+        {
+            d_socket->set_option(loption);
+        }
+        catch (const std::exception &ex)
+        {
+            GR_LOG_WARN(d_logger, "Failed to set boost::asio::socket_base::linger " + std::string(ex.what()));
+        }
 
         boost::asio::socket_base::reuse_address roption(true);
         d_socket->set_option(roption);
