diff --git a/gnuradio-runtime/include/gnuradio/logger.h.in b/gnuradio-runtime/include/gnuradio/logger.h.in
index 13fda6f6c..ec3c347f9 100644
--- a/gnuradio-runtime/include/gnuradio/logger.h.in
+++ b/gnuradio-runtime/include/gnuradio/logger.h.in
@@ -111,7 +111,7 @@ typedef unsigned short mode_t;
 
 namespace gr {
   //#warning GR logging Enabled and using std::cout
-  typedef std::string logger_ptr;
+  typedef void* logger_ptr;
 } /* namespace gr */
 
 #define GR_LOG_DECLARE_LOGPTR(logger)
