diff --git a/CMakeLists.txt b/CMakeLists.txt
index cdf4a6d..b55489a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -120,6 +120,9 @@ IF(MSVC)
     ELSE(MSVC12) #Visual Studio 12
         SET(cmake_c_compiler_version "Microsoft Visual Studio 12.0")
         SET(cmake_cxx_compiler_version "Microsoft Visual Studio 12.0")
+    ELSE(MSVC14) #Visual Studio 14
+        SET(cmake_c_compiler_version "Microsoft Visual Studio 14.0")
+        SET(cmake_cxx_compiler_version "Microsoft Visual Studio 14.0")
     ENDIF()
 ELSE()
     execute_process(COMMAND ${CMAKE_C_COMPILER} --version
diff --git a/gr-blocks/lib/peak_detector2_fb_impl.cc b/gr-blocks/lib/peak_detector2_fb_impl.cc
index dc13e66..74cf3d3 100644
--- a/gr-blocks/lib/peak_detector2_fb_impl.cc
+++ b/gr-blocks/lib/peak_detector2_fb_impl.cc
@@ -109,7 +109,7 @@ namespace gr {
             sigout[i]=d_avg;
           if(iptr[i] > d_avg * (1.0f + d_threshold_factor_rise)) {
             d_found = true;
-            d_peak_val = -(float)INFINITY;
+            d_peak_val = -(float)std::numeric_limits<float>::max();
             set_output_multiple(d_look_ahead);
             return i;
           }
diff --git a/gr-blocks/lib/peak_detector_XX_impl.cc.t b/gr-blocks/lib/peak_detector_XX_impl.cc.t
index 6846a02..33bfe62 100644
--- a/gr-blocks/lib/peak_detector_XX_impl.cc.t
+++ b/gr-blocks/lib/peak_detector_XX_impl.cc.t
@@ -70,7 +70,7 @@ namespace gr {
 
       memset(optr, 0, noutput_items*sizeof(char));
 
-      @I_TYPE@ peak_val = -(@I_TYPE@)INFINITY;
+      @I_TYPE@ peak_val = -(@I_TYPE@)std::numeric_limits<@I_TYPE@>::max();
       int peak_ind = 0;
       unsigned char state = 0;
       int i = 0;
@@ -101,7 +101,7 @@ namespace gr {
           else {
             optr[peak_ind] = 1;
             state = 0;
-            peak_val = -(@I_TYPE@)INFINITY;
+            peak_val = -(@I_TYPE@)std::numeric_limits<@I_TYPE@>::max();
             //printf("Leaving  State 1: Peak: %f  Peak Ind: %d   i: %d  noutput_items: %d\n",
             //peak_val, peak_ind, i, noutput_items);
           }

