diff --git a/cmake/Modules/GrPython.cmake b/cmake/Modules/GrPython.cmake
index 06e061e21..6abb08e10 100644
--- a/cmake/Modules/GrPython.cmake
+++ b/cmake/Modules/GrPython.cmake
@@ -81,7 +81,7 @@ macro(GR_PYTHON_CHECK_MODULE desc mod cmd have)
     message(STATUS "")
     message(STATUS "Python checking for ${desc}")
     execute_process(
-        COMMAND ${PYTHON_EXECUTABLE} -c "
+        COMMAND "${CMAKE_COMMAND}" -E env "PYTHONPATH=${CMAKE_INSTALL_PREFIX}/lib/python2.7/site-packages;$ENV{PYTHONPATH}" ${PYTHON_EXECUTABLE} -c "
 #########################################
 try:
     import ${mod}
