diff --git a/gr-fft/lib/fft.cc b/gr-fft/lib/fft.cc
index ef8158318..44b3f1ba9 100644
--- a/gr-fft/lib/fft.cc
+++ b/gr-fft/lib/fft.cc
@@ -26,6 +26,13 @@
 #include <volk/volk.h>
 #include <fftw3.h>
 
+#ifdef _MSC_VER
+#include <fcntl.h>
+#include <io.h>
+#define O_NOCTTY 0
+#define O_NONBLOCK 0
+#endif //_MSC_VER
+
 #ifdef _MSC_VER //http://www.fftw.org/install/windows.html#DLLwisdom
 static void my_fftw_write_char(char c, void *f) { fputc(c, (FILE *) f); }
 #define fftw_export_wisdom_to_file(f) fftw_export_wisdom(my_fftw_write_char, (void*) (f))
