diff --git a/cmake/msvc/config.h b/cmake/msvc/config.h
index 43792c7..b8584fd 100644
--- a/cmake/msvc/config.h
+++ b/cmake/msvc/config.h
@@ -32,7 +32,9 @@ static inline float rintf(float x){return (x > 0.0f)? floorf(x + 0.5f) : ceilf(x
 ////////////////////////////////////////////////////////////////////////
 // math constants
 ////////////////////////////////////////////////////////////////////////
+#ifndef INFINITY
 #define INFINITY HUGE_VAL
+#endif
 
 # define M_E		2.7182818284590452354	/* e */
 # define M_LOG2E	1.4426950408889634074	/* log_2 e */
