diff --git a/gr-dtv/lib/dvbt/dvbt_map_impl.cc b/gr-dtv/lib/dvbt/dvbt_map_impl.cc
index 05f6e7f..f1c077c 100644
--- a/gr-dtv/lib/dvbt/dvbt_map_impl.cc
+++ b/gr-dtv/lib/dvbt/dvbt_map_impl.cc
@@ -28,6 +28,10 @@
 #include <stdio.h>
 #include <math.h>
 
+#if defined(_MSC_VER) && (_MSC_VER < 1800)
+static inline double log2(double n) {return log(n) / log(2.0);}
+#endif
+
 namespace gr {
   namespace dtv {
 
diff --git a/gr-dtv/lib/dvbt/dvbt_symbol_inner_interleaver_impl.cc b/gr-dtv/lib/dvbt/dvbt_symbol_inner_interleaver_impl.cc
index 64d242d..3473f0f 100644
--- a/gr-dtv/lib/dvbt/dvbt_symbol_inner_interleaver_impl.cc
+++ b/gr-dtv/lib/dvbt/dvbt_symbol_inner_interleaver_impl.cc
@@ -26,6 +26,10 @@
 #include "dvbt_symbol_inner_interleaver_impl.h"
 #include <stdio.h>
 
+#if defined(_MSC_VER) && (_MSC_VER < 1800)
+static inline double log2(double n) {return log(n) / log(2.0);}
+#endif
+
 namespace gr {
   namespace dtv {
 
