diff --git a/gnuradio-runtime/include/gnuradio/unittests.h b/gnuradio-runtime/include/gnuradio/unittests.h
index 209c3ab..14c201e 100644
--- a/gnuradio-runtime/include/gnuradio/unittests.h
+++ b/gnuradio-runtime/include/gnuradio/unittests.h
@@ -26,7 +26,9 @@
 #include <string.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#ifndef _MSC_VER
 #include <unistd.h>
+#endif
 #include <string>
 
 #include <boost/filesystem/operations.hpp>
