diff --git a/gnuradio-runtime/lib/sys_paths.cc b/gnuradio-runtime/lib/sys_paths.cc
index 3bf6697..f63d773 100644
--- a/gnuradio-runtime/lib/sys_paths.cc
+++ b/gnuradio-runtime/lib/sys_paths.cc
@@ -64,10 +64,16 @@ namespace gr {
     return tmp_path();
   }
 
-  const char *userconf_path()
+  std::string __userconf_path()
   {
     boost::filesystem::path p(appdata_path());
     p = p / ".gnuradio";
+    return p.string();
+  }
+
+  const char *userconf_path()
+  {
+    static std::string p(__userconf_path());
     return p.c_str();
   }
 
