diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6312c1f..d40cb9d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,6 +56,24 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX)
     add_definitions(-Wsign-compare)
 endif()
 
+if(MSVC)
+    #get math definitions like M_PI
+    add_definitions(-D_USE_MATH_DEFINES)
+
+    #use std::min()/std::max()
+    add_definitions(-DNOMINMAX)
+
+    #needed to dynamically link boost
+    add_definitions(-DBOOST_ALL_DYN_LINK)
+
+    #export gr-rds symbols
+    add_definitions(-Dgnuradio_RDS_EXPORTS)
+
+    if ("${MSVC_VERSION}" VERSION_LESS "1900")
+        add_definitions(-D__func__=__FUNCTION__)
+    endif()
+endif()
+
 #######################################################################################################################
 # Functions & macros.  These must be defined before including subdirectories.
 
diff --git a/src/applications/gqrx/receiver.cpp b/src/applications/gqrx/receiver.cpp
index 7f82f94..4ac6674 100644
--- a/src/applications/gqrx/receiver.cpp
+++ b/src/applications/gqrx/receiver.cpp
@@ -22,7 +22,9 @@
  */
 #include <cmath>
 #include <iostream>
+#ifndef _MSC_VER
 #include <unistd.h>
+#endif
 
 #include <iostream>
 
diff --git a/src/dsp/afsk1200/filter.h b/src/dsp/afsk1200/filter.h
index 4539d01..0155449 100644
--- a/src/dsp/afsk1200/filter.h
+++ b/src/dsp/afsk1200/filter.h
@@ -30,14 +30,22 @@
 #include "filter-i386.h"
 #endif /* ARCH_I386 */
 
+#ifndef _MSC_VER
+#define DECL_UNUSED __attribute__ ((unused))
+#endif
+
+#ifndef DECL_UNUSED
+#define DECL_UNUSED
+#endif
+
 /* ---------------------------------------------------------------------- */
 
 extern inline unsigned int hweight32(unsigned int w)
-        __attribute__ ((unused));
+        DECL_UNUSED;
 extern inline unsigned int hweight16(unsigned short w)
-        __attribute__ ((unused));
+        DECL_UNUSED;
 extern inline unsigned int hweight8(unsigned char w)
-        __attribute__ ((unused));
+        DECL_UNUSED;
 
 extern inline unsigned int hweight32(unsigned int w) 
 {
@@ -64,9 +72,9 @@ extern inline unsigned int hweight8(unsigned char w)
 }
 
 extern inline unsigned int gcd(unsigned int x, unsigned int y)
-        __attribute__ ((unused));
+        DECL_UNUSED;
 extern inline unsigned int lcm(unsigned int x, unsigned int y)
-        __attribute__ ((unused));
+        DECL_UNUSED;
 
 extern inline unsigned int gcd(unsigned int x, unsigned int y)
 {
diff --git a/src/qtgui/plotter.cpp b/src/qtgui/plotter.cpp
index 9a91271..e25d542 100644
--- a/src/qtgui/plotter.cpp
+++ b/src/qtgui/plotter.cpp
@@ -28,7 +28,33 @@
  * or implied, of Moe Wheatley.
  */
 #include <cmath>
+#ifndef _MSC_VER
 #include <sys/time.h>
+#else
+
+#include <Windows.h>
+#include <cstdint>
+
+int gettimeofday(struct timeval * tp, struct timezone * tzp)
+{
+    // Note: some broken versions only have 8 trailing zero's, the correct epoch has 9 trailing zero's
+    static const uint64_t EPOCH = ((uint64_t) 116444736000000000ULL);
+
+    SYSTEMTIME  system_time;
+    FILETIME    file_time;
+    uint64_t    time;
+
+    GetSystemTime( &system_time );
+    SystemTimeToFileTime( &system_time, &file_time );
+    time =  ((uint64_t)file_time.dwLowDateTime )      ;
+    time += ((uint64_t)file_time.dwHighDateTime) << 32;
+
+    tp->tv_sec  = (long) ((time - EPOCH) / 10000000L);
+    tp->tv_usec = (long) (system_time.wMilliseconds * 1000);
+    return 0;
+}
+
+#endif
 #include <QDateTime>
 #include <QDebug>
 #include <QFont>
