diff --git a/lib/encoder_impl.cc b/lib/encoder_impl.cc
index 604c485..dfd98e1 100644
--- a/lib/encoder_impl.cc
+++ b/lib/encoder_impl.cc
@@ -18,6 +18,13 @@
 #include "config.h"
 #endif
 
+#if defined(_MSC_VER) && !defined(NOMINMAX)
+#  define NOMINMAX //allow std::min/max()
+#  if _MSC_VER < 1800 //nearbyint for vc11
+#    define nearbyint(x) ((int) ((x)+0.5))
+#  endif  
+#endif
+
 #include "encoder_impl.h"
 #include "constants.h"
 #include <gnuradio/io_signature.h>
@@ -394,7 +401,7 @@ void encoder_impl::prepare_group2(const bool AB) {
 void encoder_impl::prepare_group1a(void) {
 	std::cout << "preparing group 1" << std::endl;
 	//infoword[1] = infoword[1] | (1 << 4); // TMC in 8A
-	infoword[2] = (0b10000000 << 8) | 0xE0;
+	infoword[2] = (0x80 << 8) | 0xE0;
 	infoword[3] = 0; // time
 }
 
@@ -440,7 +447,7 @@ void encoder_impl::prepare_group4a(void) {
 	infoword[1]=infoword[1]|((mjd>>15)&0x3);
 	infoword[2]=(((mjd>>7)&0xff)<<8)|((mjd&0x7f)<<1)|((h>>4)&0x1);
 	infoword[3]=((h&0xf)<<12)|(((m>>2)&0xf)<<8)|((m&0x3)<<6)|
-		((toffset>0?0:1)<<5)|((abs(toffset*2))&0x1f);
+		((toffset>0?0:1)<<5)|((int(abs(toffset*2)))&0x1f);
 }
 
 // for now single-group only
