diff --git a/CMakeLists.txt b/CMakeLists.txt
index a5dd8f5..b45581c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -124,7 +124,7 @@ endif(APPLE)
 find_package(Doxygen)
 
 MESSAGE(STATUS "Configuring GNU Radio C++ Libraries...")
-set(GR_REQUIRED_COMPONENTS RUNTIME)
+set(GR_REQUIRED_COMPONENTS RUNTIME PMT)
 set(MIN_GR_VERSION "3.7.8")
 find_package(Gnuradio REQUIRED)
 if("${Gnuradio_VERSION}" VERSION_LESS MIN_GR_VERSION)
