diff --git a/CMakeLists.txt b/CMakeLists.txt
index a2f7159..ee7741d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,7 +30,7 @@ ELSE (WIN32)
   ENDIF (APPLE)
 ENDIF (WIN32)
 
-SET ( LOG4CPP_LIBRARY_NAME "orocos-log4cpp" )
+SET ( LOG4CPP_LIBRARY_NAME "log4cpp" )
 
 ADD_LIBRARY ( ${LOG4CPP_LIBRARY_NAME} SHARED
   src/Appender.cpp
@@ -132,18 +132,18 @@ SET(prefix "${CMAKE_INSTALL_PREFIX}")
 SET(exec_prefix "\${prefix}")
 SET(log4cpp_cflags "")
 SET(log4cpp_libs "${LOG4CPP_LIBS}")
-SET(includedir "\${prefix}/include/orocos")
+SET(includedir "\${prefix}/include")
 SET(libdir "\${prefix}/lib")
-CONFIGURE_FILE(log4cpp.pc.in ${CMAKE_CURRENT_BINARY_DIR}/orocos-log4cpp.pc @ONLY)
+CONFIGURE_FILE(log4cpp.pc.in ${CMAKE_CURRENT_BINARY_DIR}/log4cpp.pc @ONLY)
 
 INSTALL(
-  FILES ${CMAKE_CURRENT_BINARY_DIR}/orocos-log4cpp.pc
+  FILES ${CMAKE_CURRENT_BINARY_DIR}/log4cpp.pc
   DESTINATION lib/pkgconfig
   )
 
 INSTALL (
   DIRECTORY include/log4cpp
-  DESTINATION include/orocos
+  DESTINATION include
   PATTERN "config.h.in" EXCLUDE
   PATTERN ".svn" EXCLUDE
   PATTERN "*.am" EXCLUDE
@@ -153,13 +153,13 @@ INSTALL (
 
 INSTALL (
   FILES ${CMAKE_CURRENT_BINARY_DIR}/include/log4cpp/config.h
-  DESTINATION include/orocos/log4cpp
+  DESTINATION include/log4cpp
   )
 
 INSTALL(TARGETS ${LOG4CPP_LIBRARY_NAME}
     LIBRARY DESTINATION lib
     ARCHIVE DESTINATION lib
-    RUNTIME DESTINATION lib)
+    RUNTIME DESTINATION bin)
 
 INSTALL(FILES package.xml DESTINATION share/log4cpp)
 
