diff --git a/lib/soapy/soapy_source_c.cc b/lib/soapy/soapy_source_c.cc
index 4bd3aa3..873c4f0 100644
--- a/lib/soapy/soapy_source_c.cc
+++ b/lib/soapy/soapy_source_c.cc
@@ -281,7 +281,8 @@ void soapy_source_c::set_dc_offset( const std::complex<double> &offset, size_t c
 
 void soapy_source_c::set_iq_balance_mode( int mode, size_t chan )
 {
-    throw std::runtime_error("soapy_source_c::set_iq_balance_mode() not supported");
+    if (mode != 0)
+        throw std::runtime_error("soapy_source_c::set_iq_balance_mode() not supported");
 }
 
 void soapy_source_c::set_iq_balance( const std::complex<double> &balance, size_t chan )
