diff --git a/lib/rtl_tcp/rtl_tcp_source_f.h b/lib/rtl_tcp/rtl_tcp_source_f.h
index d1d918c..84ac57a 100644
--- a/lib/rtl_tcp/rtl_tcp_source_f.h
+++ b/lib/rtl_tcp/rtl_tcp_source_f.h
@@ -42,7 +42,10 @@ typedef char* optval_t;
 typedef void* optval_t;
 #endif
 
-#define ssize_t int
+#ifdef _MSC_VER
+#include <cstddef>
+typedef ptrdiff_t ssize_t;
+#endif //_MSC_VER
 
 /* copied from rtl sdr */
 enum rtlsdr_tuner {
