diff --git a/spuce/typedefs.h b/spuce/typedefs.h
index 495c00e..a91d3fd 100644
--- a/spuce/typedefs.h
+++ b/spuce/typedefs.h
@@ -1,6 +1,7 @@
 #pragma once
 // Copyright (c) 2015 Tony Kirke.  Boost Software License - Version 1.0  (http://www.opensource.org/licenses/BSL-1.0)
 #define _USE_MATH_DEFINES
+#include <math.h>
 // Putting std::complex here allows possible replacement with custom complex type later
 #include <complex>
 #include "complex_operators.h"
 
 