diff --git a/spuce/filters/iir_coeff.cpp b/spuce/filters/iir_coeff.cpp
index 53b6880..cdfd027 100644
--- a/spuce/filters/iir_coeff.cpp
+++ b/spuce/filters/iir_coeff.cpp
@@ -152,9 +152,9 @@ void iir_coeff::make_band(float_type c) {
     b_tf.pop_back();
     b_tf.pop_back();
     // Actual polynomials will be real only
-    std::vector<float_type> p = {1, 0, 0};
+    float_type p[] = {1, 0, 0};
     // For bandpass zeros are always like this, even for different center frequencies!!!
-    std::vector<float_type> z = {1, 0, -1};
+    float_type z[] = {1, 0, -1};
     // Page 29, 111, 112
     if (lpf != filter_type::bandpass) {
       z[1] = -2 * c0;
@@ -168,8 +168,8 @@ void iir_coeff::make_band(float_type c) {
     p[1] = -2 * real(p0p);
     p[2] = std::real(norm(p0p));
     // Adds back last two elements with new convolved result
-    b_tf = convolve(z, b_tf);
-    a_tf = convolve(p, a_tf);
+    b_tf = convolve(std::vector<float_type>(z, z+3), b_tf);
+    a_tf = convolve(std::vector<float_type>(p, p+3), a_tf);
   }
 }
 
diff --git a/spuce/filters/remez_estimate.cpp b/spuce/filters/remez_estimate.cpp
index 1552327..a28c8b2 100644
--- a/spuce/filters/remez_estimate.cpp
+++ b/spuce/filters/remez_estimate.cpp
@@ -61,7 +61,7 @@ float_type remez_estimate_atten(size_t num_taps, float_type trans_bw, float_type
 */
 size_t remez_estimate_num_taps(float_type trans_bw, float_type ripple, float_type atten) {  
   size_t num_taps;
-  std::vector<float_type> AA = {-4.278e-01,-4.761e-01,0,
+  float_type AA[] = {-4.278e-01,-4.761e-01,0,
                                 -5.941e-01,7.114e-02,0,
                                 -2.660e-03,5.309e-03,0};
 
@@ -71,7 +71,7 @@ size_t remez_estimate_num_taps(float_type trans_bw, float_type ripple, float_typ
   float_type d2 = -atten/20.0;
 
   std::vector<float_type> AD(3);
-  std::vector<float_type> dd = {1, d2, d2*d2};
+  float_type dd[] = {1, d2, d2*d2};
 
   size_t off=0;
   for (size_t j=0;j<3;j++) {
@@ -106,7 +106,7 @@ size_t remez_estimate_num_taps(float_type trans_bw, float_type ripple, float_typ
 */
 size_t remez_estimate_num_taps(float_type edge1, float_type ripple1, float_type edge2, float_type ripple2) {  
   size_t num_taps;
-  std::vector<float_type> AA = {-4.278e-01,-4.761e-01,0,
+  float_type AA[] = {-4.278e-01,-4.761e-01,0,
                                 -5.941e-01,7.114e-02,0,
                                 -2.660e-03,5.309e-03,0};
 
@@ -117,7 +117,7 @@ size_t remez_estimate_num_taps(float_type edge1, float_type ripple1, float_type
   float_type d2 = -ripple2/20.0;
 
   std::vector<float_type> AD(3);
-  std::vector<float_type> dd = {1, d2, d2*d2};
+  float_type dd[] = {1, d2, d2*d2};
 
   size_t off=0;
   for (size_t j=0;j<3;j++) {
diff --git a/spuce/filters/window.cpp b/spuce/filters/window.cpp
index 2530874..aa0211e 100644
--- a/spuce/filters/window.cpp
+++ b/spuce/filters/window.cpp
@@ -7,6 +7,9 @@
 #include <spuce/filters/fir_inv_dft.h>
 #include <iostream>
 #define TWOPI 2*M_PI
+#if defined(_MSC_VER) && (_MSC_VER < 1800)
+#define acosh(x) log((x) + sqrt((x)*(x) - 1))
+#endif
 namespace spuce {
 
 inline std::complex<float_type> expj(float_type x) { return (std::complex<float_type>(std::cos(x), std::sin(x))); }
 