diff --git a/host/lib/utils/gain_group.cpp b/host/lib/utils/gain_group.cpp
index e907a65..0f56eb8 100644
--- a/host/lib/utils/gain_group.cpp
+++ b/host/lib/utils/gain_group.cpp
@@ -51,6 +51,7 @@ static bool compare_by_step_size(
  * \return a multiple of step approximating num
  */
 template <typename T> static T floor_step(T num, T step, T e = T(0.001)){
+    if (num < T(0)) return step*int(num/step - e);
     return step*int(num/step + e);
 }

