diff --git a/apps/CMakeLists.txt b/apps/CMakeLists.txt
index 74ed18d..58d05ae 100644
--- a/apps/CMakeLists.txt
+++ b/apps/CMakeLists.txt
@@ -20,6 +20,7 @@
 ########################################################################
 if(MSVC)
     include_directories(${CMAKE_SOURCE_DIR}/cmake/msvc)
+    add_definitions(-DHAVE_CONFIG_H)
 endif(MSVC)
 
 include_directories(
diff --git a/lib/qa_utils.cc b/lib/qa_utils.cc
index 96b5170..400c3e8 100644
--- a/lib/qa_utils.cc
+++ b/lib/qa_utils.cc
@@ -1,3 +1,7 @@
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "qa_utils.h"
 
 #include <boost/foreach.hpp>
