diff --git a/cmake/msvc/config.h b/cmake/msvc/config.h
index 43792c7..b72c442 100644
--- a/cmake/msvc/config.h
+++ b/cmake/msvc/config.h
@@ -28,6 +28,7 @@ static inline long long llrint(double x){return (long long)(x > 0.0 ? x + 0.5 :
 static inline long long llrintf(float x){return (long long)(x > 0.0f ? x + 0.5f : x - 0.5f);}
 static inline double rint(double x){return (x > 0.0)? floor(x + 0.5) : ceil(x - 0.5);}
 static inline float rintf(float x){return (x > 0.0f)? floorf(x + 0.5f) : ceilf(x - 0.5f);}
+static inline double log2(double n) {return log(n) / log(2.0);}
 
 ////////////////////////////////////////////////////////////////////////
 // math constants
