diff --git a/cmake/msvc/config.h b/cmake/msvc/config.h
index 43792c7..6729a1f 100644
--- a/cmake/msvc/config.h
+++ b/cmake/msvc/config.h
@@ -29,6 +29,15 @@ static inline long long llrintf(float x){return (long long)(x > 0.0f ? x + 0.5f
 static inline double rint(double x){return (x > 0.0)? floor(x + 0.5) : ceil(x - 0.5);}
 static inline float rintf(float x){return (x > 0.0f)? floorf(x + 0.5f) : ceilf(x - 0.5f);}
 
+#if _MSC_VER < 1800
+static inline double log2(double n) {return log(n) / log(2.0);}
+
+static inline float copysignf(float x, float y)
+{
+  return (float)_copysign((double)x, (double)y);
+}
+#endif
+
 ////////////////////////////////////////////////////////////////////////
 // math constants
 ////////////////////////////////////////////////////////////////////////
