diff --git a/CMakeLists.txt b/CMakeLists.txt
index aa40e18..34846f0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -688,7 +688,7 @@ foreach(readme ${readme-docs})
 
   if(NOT ZMQ_BUILD_FRAMEWORK)
     if(MSVC)
-      install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${readme}.txt DESTINATION .)
+      install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${readme}.txt DESTINATION etc/zmq)
     else()
       install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${readme}.txt DESTINATION etc/zmq)
     endif()
